go.mod

```diff
@@ -4,13 +4,13 @@ require (
 	github.com/btcsuite/btcd v0.20.1-beta
 	github.com/btcsuite/btclog v0.0.0-20170628155309-84c8d2346e9f
 	github.com/golang/protobuf v1.3.3
-	github.com/grpc-ecosystem/grpc-gateway v1.10.0
+	github.com/grpc-ecosystem/grpc-gateway v1.12.2
```

Comment:
I wonder why we keep having an old version of grpc-gateway in lnd.

Reply:
We can't update `grpc-gateway` in lnd because it has a problem with the `rpc.proto` containing two services. See https://github.com/lightningnetwork/lnd/pull/3650. But that will be fixed with my upcoming proto overhaul PR (lightningnetwork/lnd#4141).

The reason why I would be cautious when updating these in other projects is that if the version used as a dependency and the version used to compile the protos is different, it _could_ potentially lead to problems. Because on a system you only have one set of `protoc` and `genproto` installed (with the versions pinned to what is needed in lnd), you normally don't switch them when compiling these protos. So there could be a difference in theory if the versions drift apart too much.

Reply:
I think there may be a historical reason for this, but also unsure.